Plugins/Json/src/DetectorVolumeJsonConverter.cpp

```diff
@@ -0,0 +1,186 @@
+    const GeometryContext& gctx, const Experimental::DetectorVolume& volume,
+    const std::vector<const Experimental::DetectorVolume*>& detectorVolumes,
+    const std::vector<const Experimental::Portal*>& portals,
+    const Options& options) {
+  nlohmann::json jVolume;
+  jVolume["name"] = volume.name();
+  jVolume["transform"] = Transform3JsonConverter::toJson(
+      volume.transform(gctx), options.transformOptions);
+  jVolume["bounds"] = VolumeBoundsJsonConverter::toJson(volume.volumeBounds());
+  // Write the surfaces
+  nlohmann::json jSurfaces;
+  std::for_each(
+      volume.surfaces().begin(), volume.surfaces().end(), [&](const auto& s) {
+        jSurfaces.push_back(
+            SurfaceJsonConverter::toJson(gctx, *s, options.surfaceOptions));
+      });
+  jVolume["surfaces"] = jSurfaces;
+  // And its surface navigation delegates
+  nlohmann::json jSurfacesDelegate =
+      IndexedSurfacesJsonConverter::toJson(volume.surfaceCandidatesUpdator());
+  jVolume["surface_navigation"] = jSurfacesDelegate;
+
+  // Write the sub volumes
+  nlohmann::json jVolumes;
+  std::for_each(
+      volume.volumes().begin(), volume.volumes().end(), [&](const auto& v) {
+        jVolumes.push_back(toJson(gctx, *v, detectorVolumes, portals, options));
+      });
+  jVolume["volumes"] = jVolumes;
+
+  // Write the portals if pre-converted as link
+  nlohmann::json jPortals;
+  if (not portals.empty()) {
+    for (const auto& p : volume.portals()) {
+      auto it = std::find(portals.begin(), portals.end(), p);
+      if (it != portals.end()) {
+        jPortals.push_back(std::distance(portals.begin(), it));
+      } else {
+        throw std::runtime_error("Portal not found in the list of portals");
+      }
+    }
+    jVolume["portal_links"] = jPortals;
+  } else {
+    for (const auto& p : volume.portals()) {
+      nlohmann::json jPortal = PortalJsonConverter::toJson(
+          gctx, *p, detectorVolumes, options.portalOptions);
+      jPortals.push_back(jPortal);
+    }
+    jVolume["portals"] = jPortals;
+  }
+  return jVolume;
+}
+
+nlohmann::json Acts::DetectorVolumeJsonConverter::toJsonDetray(
+    const GeometryContext& gctx, const Experimental::DetectorVolume& volume,
+    const std::vector<const Experimental::DetectorVolume*>& detectorVolumes,
+    const Options& options) {
+  nlohmann::json jVolume;
+  jVolume["name"] = volume.name();
+
+  // Write the transform - path them with defaults
+  jVolume["transform"] = Transform3JsonConverter::toJson(
+      volume.transform(gctx), options.transformOptions);
+  jVolume["bounds"] = VolumeBoundsJsonConverter::toJson(volume.volumeBounds());
+  auto volumeBoundsType = volume.volumeBounds().type();
+  if (volumeBoundsType == VolumeBounds::BoundsType::eCylinder) {
+    jVolume["type"] = 0u;
+  } else if (volumeBoundsType == VolumeBounds::BoundsType::eCuboid) {
+    jVolume["type"] = 4u;
+  } else {
+    throw std::runtime_error("Unsupported volume bounds type");
```

Comment:
Technically not entirely true, practically we have never tried anything else than cylinders and cuboids.
detray volume types
```

/// Shape of a detector volume.
///
/// cylinder: cylinder and two disc portals (shape: cylinder2D or cylinder3D).
/// cone: a cone portal and a disc portal (shape: missing).
/// rectangle: six rectangular portals that form a box (shape: cuboid3D).
/// trapezoid: six trapezoid portals (shape: cuboid3D).
/// cuboid: general cuboid form, excluding the previous ones (shape: cuboid3D).
enum class volume_id : std::uint_least8_t {
    e_cylinder = 0u,
    e_rectangle = 1u,
    e_trapezoid = 2u,
    e_cone = 3u, // this is indeed not implemented
    e_cuboid = 4u
};
```

Reply:
Ok, once we have more, and more tested, we update those.